test_get_nam.sh

```diff
@@ -37,28 +45,39 @@ sed \
     -e "s/\"%NULLGETNAMFILE%\"/null/" \
     -e "s/%NULLLASTUPDATER%/$THIS/" \
     -e "s/%NULLLASTUPDATETIME%/$DATETIME/" \
-     < $SCRIPTDIR/$namTemplateName \
-     > $filledNamTemplateName
+        < $SCRIPTDIR/$namTemplateName \
+        > $filledNamTemplateName \
 # getting nam status
-latestCycle=$($SCRIPTDIR/get_nam_status.pl  \
-             < $filledNamTemplateName       \
-             | $SCRIPTDIR/latest.pl         \
-            )
-# selecting
-thisCycle=$(sed "s/%NULLFORECASTCYCLE%/$arrFORECASTCYCLE/" \
-            < get_nam_status.pl.json                       \
-            | $SCRIPTDIR/select_nam_nowcast.pl             \
-            | $SCRIPTDIR/latest.pl)
-#  n o w c a s t
-stage=NOWCAST
-sed \
-    -e "s/%NULLSTAGE%/$stage/"  \
-    -e "s/%NULLSCRIPTDIR%/$escSCRIPTDIR/" \
-    -e "s/%NULLNAMDATAPATH%/$escInstanceNamDir/" \
-     < select_nam_nowcast.pl.json \
-     | $SCRIPTDIR/get_nam_data.pl \
-     > /dev/null
-# for converting to ascii win/pre (owi) format
+get_nam_status.pl < $filledNamTemplateName > get_nam_status.pl.json
+if [[ $? != 0 ]]; then
+    echo "ERROR: Failed to get status of NAM cycles with get_nam_status.pl."
+    exit
```

Comment:
```suggestion
err=$?
if [[ $err != 0 ]]; then
    echo "ERROR: Failed to get status of NAM cycles with get_nam_status.pl."
    exit $err
```

Do we need to pass the error code to the parent process calling this script?

Reply:
yes